src/figuresClasses.ts

```diff
@@ -1,19 +1,85 @@
-export interface Figure {
+// import { CpuInfo } from "os";
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
 }
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle {
-
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea():number,
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
 
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (
+      this.a + this.b <= this.c
+      || this.b + this.c <= this.a
+      || this.c + this.a <= this.b
+    ) {
+      throw new Error('This is not a triangle');
+    }
```

Comment:
In my opinion, better to put this check to function and just call it in constructor

Reply:
agree, will be better to create separate function for this check